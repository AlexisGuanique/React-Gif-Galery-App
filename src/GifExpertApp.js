import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

// Creacion del componente
export const GifExpertApp = () => {
    
    
    // De esta manera ceramos la lista de elementos para poder luego manipularla con el set categories
    const [categories, setCategories] = useState(['Dragon Ball Z'])
    

    return (
        <>
            <h1>Gif Galery App</h1>

            {/* Le pasamos setCategories como una propiedad a nuestro componente AddCategory */}
            <AddCategory setCategories = { setCategories } />
            <hr />

            <ol>
                {
                    // Mapeamos la lista categories para poder recorres elemento por elemento
                    categories.map( category => 
                        <GifGrid 
                            key= { category }
                            category= { category }
                        />
                    )
                }
            </ol>
        </>
    )
};


