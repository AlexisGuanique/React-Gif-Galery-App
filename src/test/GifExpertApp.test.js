import React from "react";
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import { GifExpertApp } from "../GifExpertApp";


describe('Pruebas con el componente <GifExpertApp />', () => {

    const wrapper = shallow(<GifExpertApp />)   

    test('Debe mostrar el componente <GifExpertApp /> correctamente', () => {

        expect( wrapper ).toMatchSnapshot()
        
    });

    test('Debe de mostrar una lista de categorias', () => {

        const categories = ['Dragon Ball Z', 'Samurai X'];

        const wrapper = shallow(<GifExpertApp defaultCategories={ categories } />);

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length);
    });

});