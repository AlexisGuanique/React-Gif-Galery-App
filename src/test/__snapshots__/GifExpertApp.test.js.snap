// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con el componente <GifExpertApp /> Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h1>
    Gif Galery App
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Dragon Ball Z"
      key="Dragon Ball Z"
    />
    <GifGrid
      category="Samurai X"
      key="Samurai X"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas con el componente <GifExpertApp /> Debe mostrar el componente <GifExpertApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Gif Galery App
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
